src/http2_connection.erl

```diff
@@ -1529,10 +1540,9 @@ maybe_hpack(Continuation, Conn) ->
 %% * Only acceptable psuedoheaders are:
 %%     :method, :scheme, :authority, :path,
 
-good_req_headers(Headers) ->
+good_headers(Type, Headers) ->
```

Comment:
Originally it was something like

```erlang
validate_pseudos(Headers) andalso no_upper_names(Headers)
```

but I squeaked a little performance optimization out by only running no_upper_names on Headers' tail after the pseudos were already looked at.

This function effectively frees me to write true/false validators for the Headers list and wrap it up in a compression error for the `hpack_maybe` function.